packages/amazonq/src/app/inline/completion.ts

```diff
@@ -435,6 +493,14 @@ ${itemLog}
             // the user typed characters from invoking suggestion cursor position to receiving suggestion position
             const typeahead = document.getText(new Range(position, editor.selection.active))
 
+            // Check if an edit suggestion is currently active - if so, discard completion suggestions
+            if (this.isEditSuggestionActive()) {
+                this.batchDiscardTelemetryForEditSuggestion(items, session)
+                this.sessionManager.clear()
+                logstr += `- completion suggestions discarded due to active edit suggestion`
+                return []
+            }
+
```

Comment:
>A newer edit suggestion arriving from the server. In such cases, the currently displayed edit suggestion is hidden from the user with a DISCARD state, and replaced with the newer edit suggestion.

Reply:
+1, 
```
if (isEditSuggestionActive && !isInlineEdit) {
    return
}
```

Reply:
the other way i think you can also do 
```
if (!isTriggerByDeletion && !request.partialResultToken && !isEditSuggestionActive) {
    promise.push(completionPromise)
}
```

here in recommendationService https://github.com/aws/aws-toolkit-vscode/blob/master/packages/amazonq/src/app/inline/recommendationService.ts#L135-L141

Reply:
since the only scenario where 2 overlapping suggestion will happen is 
1. when there is an Edit being shown
2. the Edit is not dismissed by users
3. users again start typing and trigger "Completion"

Reply:
Implemented Will's second solution - added !EditSuggestionState.isEditSuggestionActive() to the condition in recommendationService.ts

Reply:
Removed